### ISSUE TITLE
Suggestion for documentation improvement

### ISSUE THREAD

**Issue by Dev-iL** [evant/binding-collection-adapter#50]
I'd like to suggest that the readme is expanded to list the benefits of using this library versus the features provided by the core databinding library. As it stands, the purpose of the library isn't very clear. 

If the main claim is that "it makes things easier", perhaps a "before and after" example would be beneficial.
If it adds functionality (such as allowing to choose from multiple `itemView`s) this should be stressed, imho.

(The other day I was trying to convince a colleague that he should be using this library, but he insisted that one needs nothing else besides the default databinding library to create bindings for `RecyclerView` etc. )

**Comment by shorthander** [comment]
And for me it would be great to have a documentation how to use a onClick handler in a recyclerview item. I do not get the link between the viewmodel (item) and the xml. Where do I bind the listeners in a recycler view item? This is a open point for me.

**Comment by evant** [comment]
Fair enough, documentation can always be better. Sure you can write you own adapter with databinding, but won't they all look pretty much the same? There are also some subtile implementation details that you might easily miss (https://realm.io/news/data-binding-android-boyar-mount/#recyclerview-and-data-binding-2214javascriptpresentzchangechapter071true does a good job at covering this).

The goal for this lib is to provide a similar separation and update mechanism to collections as databinding does for plain views. You can simply work on a collection of data and have it update the ui with as minimal glue code as possible.
